### PR TITLE
Increased the precision of validation metric to be saved in the file names.

### PR DESCRIPTION
**Pull request by VahidooX**
Currently we save the value of the validation metric into the name of the checkpoints upto 2 numbers after the decimal point. For some models like ASR where we have WER and numbers are less than 1, it can cause an issue. The experiment manager cannot find the right topk checkpoints.

### CODE REVIEW

**Comment by blisc** [comment]
> Currently we save the value of the validation metric into the name of the checkpoints upto 2 numbers after the decimal point. For some models like ASR where we have WER and numbers are less than 1, it can cause an issue. The experiment manager cannot find the right topk checkpoints.

If you are measuring something like WER, doesn't it make more sense to just multiply your metric by 100?

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -798,7 +798,7 @@ def configure_checkpointing(
     if params.dirpath is None:
         params.dirpath = Path(log_dir / 'checkpoints')
     if params.filename is None:
-        params.filename = f'{name}--{{{params.monitor}:.2f}}-{{epoch}}'
+        params.filename = f'{name}--{{{params.monitor}:.4f}}-{{epoch}}'
```
Comment (blisc) [inline nemo/utils/exp_manager.py:801]: This is fine with me. @titu1994 , any thoughts?

**Comment by VahidooX** [comment]
> 
> 
> > Currently we save the value of the validation metric into the name of the checkpoints upto 2 numbers after the decimal point. For some models like ASR where we have WER and numbers are less than 1, it can cause an issue. The experiment manager cannot find the right topk checkpoints.
> 
> If you are measuring something like WER, doesn't it make more sense to just multiply your metric by 100?

You are right, this can be another solution. I just thought maybe this way can be useful for other tasks too?

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (titu1994) [reply nemo/utils/exp_manager.py:801]: This is good. Multiplying by 100 is problematic cause you have to remember the scale up in downstream stuff.

**Review by titu1994 — APPROVED** [review]
I like this solution. It's more generic and doesn't depend on individual metrics to scale up and down.

**Review by blisc — APPROVED** [review]

**Pull request merged**